app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
+      const token = firebase.auth().createCustomToken(req.query.uid);
```

Comment:
Could i impersonate a user this way if i modify the req params? Remember the bearer token is per user on the api, so it's all you need to ID a user.